f8a_worker/workers/stackaggregator_v2.py

```diff
@@ -153,20 +153,29 @@ def aggregate_stack_data(stack, manifest_file, ecosystem, deps, manifest_file_pa
     license_analysis, dependencies = perform_license_analysis(license_score_list, dependencies)
     stack_license_conflict = len(license_analysis.get('f8a_stack_licenses', [])) == 0
 
+    unknown_dependencies = [{
+        'name': package,
+        'version': version} for package, version in {
+            (dependency['package'], dependency['version']) for dependency in deps
+         }.difference({
+             (dependency['name'], dependency['version']) for dependency in dependencies}
+                      )
+    ]
```

Comment:
This looks really hard to read. Could you please rewrite it to some other form? Thanks in advance!